typing/typemod.ml

```diff
@@ -707,6 +764,24 @@ let simplify_signature sg =
   let (sg, _) = aux sg in
   sg
 
+let remove_inserted_modtype mty =
+  let rec aux = function
+    | Mty_signature sg ->
+        let sg =
+          List.filter (
+            function
+            | Sig_module({Ident.name}, _, _) when open_generated name -> false
+            | _ -> true) sg in
+        Mty_signature sg
+    | Mty_functor (id, mty_arg, mty_res) ->
+        Mty_functor (id,
+        (match mty_arg with
+         | None -> None
+         | Some mty -> Some (aux mty)),
+        (aux mty_res))
+    | mty -> mty in
```

Comment:
It would be better to list the two remaining cases explicitly.

Reply:
done